src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,90 @@
+const expensesServices = require('../services/expenses.services');
+const usersServices = require('../services/users.services');
+
+const { asyncHandler } = require('../utils/asyncHandler.js');
+const { isExpenseValid } = require('../utils/expenses.js');
+const { getErrorWithStatus } = require('../utils/getError.js');
+const { getValidId } = require('../utils/validation.js');
+
+const getExpenses = async (req, res) => {
+  const { categories, userId, from, to } = req.query;
+
+  let categoriesArray = categories;
+
+  if (categories) {
+    categoriesArray = Array.isArray(categories) ? categories : [categories];
```

Comment:
The `categoriesArray` logic might not work as intended if multiple categories are provided as a comma-separated string. Consider splitting the string by commas to ensure it is always treated as an array.